core/chains/evm/client/chain_client.go

```diff
@@ -60,6 +60,9 @@ type Client interface {
 	HeadByNumber(ctx context.Context, n *big.Int) (*evmtypes.Head, error)
 	HeadByHash(ctx context.Context, n common.Hash) (*evmtypes.Head, error)
 	SubscribeNewHead(ctx context.Context, ch chan<- *evmtypes.Head) (ethereum.Subscription, error)
+	// LatestFinalizedBlock - returns the latest finalized block as it's returned from an RPC.
+	// CAUTION: Do not use this or any other ChainClient method to determine the latest finalized block.
+	// Direct use might cause local finality violations. It's highly recommended to use HeadTracker to get latest finalized block.
```

Comment:
We need a softer language here. Exposing this method and saying don't use it, is not ideal. Maybe something like this:
```suggestion
	// LatestFinalizedBlock - returns the latest finalized block as it's returned from an RPC.
	// CAUTION: Using this method might cause local finality violations. It's highly recommended
	// to use HeadTracker to get latest finalized block.
```